compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -911,8 +911,13 @@ impl<'a> Parser<'a> {
         has_parens: bool,
         modifiers: BoundModifiers,
     ) -> PResult<'a, GenericBound> {
-        let lifetime_defs = self.parse_late_bound_lifetime_defs()?;
-        let path = self.parse_path(PathStyle::Type)?;
+        let mut lifetime_defs = self.parse_late_bound_lifetime_defs()?;
+        let mut path = self.parse_path(PathStyle::Type)?;
+
+        if self.token == TokenKind::OpenDelim(Delimiter::Parenthesis) {
```

Comment:
```suggestion
        if self.may_recover() && self.token == TokenKind::OpenDelim(Delimiter::Parenthesis) {
```

To avoid stepping into #103534 territory.

Reply:
Great. Thanks!